contracts/src/BackedCommunityTokenV1.sol

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.12;
+
+import {IBackedCommunityTokenV1} from "./interfaces/IBackedCommunityTokenV1.sol";
+import {IBackedCommunityTokenDescriptorV1} from "./interfaces/IBackedCommunityTokenDescriptorV1.sol";
+import {BackedCommunityStorageV1} from "./BackedCommunityStorageV1.sol";
+import {IERC721} from "../lib/openzeppelin-contracts/contracts/token/ERC721/IERC721.sol";
+import {ERC721SoulboundUpgradeable} from "./ERC721SoulboundUpgradeable.sol";
+
+contract BackedCommunityTokenV1 is
+    BackedCommunityStorageV1,
+    ERC721SoulboundUpgradeable,
+    IBackedCommunityTokenV1
+{
+    // ==== modifiers ====
+    modifier onlyAdmin() {
+        require(msg.sender == admin, "BackedCommunityTokenV1: not admin");
+        _;
+    }
+
+    // ==== constructor ====
+    function initialize(address _admin, address _descriptor)
+        public
+        initializer
+    {
+        __ERC721Soulbound_init("BackedCommunity", "BACKED");
+        admin = _admin;
+        descriptor = _descriptor;
+    }
+
+    // ==== state changing external functions ====
+
+    function addCategory(string memory displayName)
+        external
+        override
+        onlyAdmin
+    {
+        uint256 currentCategoryId = categoryCount++;
+        categoryIdToDisplayName[currentCategoryId] = displayName;
+    }
+
+    function addSpecialAccessory(address artContract)
+        external
+        override
+        onlyAdmin
+    {
+        uint256 currentAccessoryId = specialAccessoryCount++;
+        accessoryIdToArtContract[currentAccessoryId] = artContract;
+    }
+
+    function setCategoryScores(CategoryScoreChange[] memory changes)
+        external
+        override
+        onlyAdmin
+    {
+        for (uint256 i = 0; i < changes.length; i++) {
+            _setCategoryScore(changes[i]);
+        }
+    }
+
+    function unlockAccessories(AccessoryUnlockChange[] memory changes)
+        external
+        override
+        onlyAdmin
+    {
+        for (uint256 i = 0; i < changes.length; i++) {
+            _unlockAccessory(changes[i]);
+        }
+    }
+
+    function setEnabledAccessory(uint256 accessoryId) external override {
+        require(
```

Comment:
Not sure how this could work with traits that are admin set, could maybe have a bool in that trait struct as to whether is admin permission based and then we look up in a special mapping. cc @jringenberg for clarity on whether we have XP based traits but I think we wanted to leave the door open?

Reply:
i think you're right, will update